### DISCUSSION TITLE
espnow communication break after few meter ( approx. 4 meter )

### DISCUSSION THREAD

**Discussion by aqildad-create** [pschatzmann/arduino-audio-tools#835] [Q&A] [answered]
I have esp32 wroom with out antenna  and power up using laptop usb port . 

I try below
1: i2s -> decoder sbc -> espnow -> i2s out 
2: sd -> espnow ->i2s out  
and some other tests but all of them communication break after few meter  and if i bring esp board again close enough  then communication again start properly and getting good quality audio .

what is the reason ? i don't think its power issue because if its power issue then esp32 may reset  .   only communication break greater then 4,5 meters distance

i cant figure out what cause this  . as far  my understanding espnow distance is not that much low  their is some other issue which  cant figure out . please suggest

**Comment by pschatzmann** [comment, marked as answer]
Maybe some hardware defect: Did you try to replace the devices ?
You can also try to lower the transmission speed...

**Reply by aqildad-create** [reply]
I replace transmitter and receiver both but the issue  is same . 

further see in this link 
https://blogs.ntu.edu.sg/ps9888-2020-g16/2020/07/11/esp-now-data-transfer/
Almost all packets were lost at WIFI_PHY_RATE_54M. so may be distance has some link  with wifi phy rate . I know  you used  WIFI_PHY_RATE_2M_S .may be it has some effects 
![image](https://github.com/pschatzmann/arduino-audio-tools/assets/65456998/edbfe0e3-988b-41bf-b92c-d8dff7cde105)

My current transmission rate / speed was 85,000bytes /seconds which is not very high  .i use this(esp_wifi_config_espnow_rate(WIFI_IF_STA, WIFI_PHY_RATE_2M_S);)  setting which you are also using in your API . 

With lower distance  I can get twice or thrice  of above data rate but once  increase the distance to 4,5 meters communication break even at rate of 85,000bytes /seconds .

 I also notice even if I stand in between 2esps the success transmission rate  also decrease . 

In this link people are getting success results more then 100 meter but their transmission rate are very low but for audio we need minimum approx. rate of 85000bytes /sec to get good quality audio . 
https://randomnerdtutorials.com/esp-now-esp32-arduino-ide/
![image](https://github.com/pschatzmann/arduino-audio-tools/assets/65456998/4c3a111e-7bca-48ba-880f-748c9801b653)
